code/clue.js

```diff
@@ -1,59 +1,325 @@
+      document.getElementById(
+        "killerDescription"
+      ).innerHTML = `${mystery.killer.description}`;
+      document.getElementById(
+        "killerFavouriteWeapon"
+      ).innerHTML = `Favourite Weapon: ${mystery.killer.favouriteWeapon}`;
+    }, 2000);
+    killerPicked = true;
+  }
+};
 
 // CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// This function will be invoked when you click on the weapon card.
+const pickWeapon = (event) => {
+  event.preventDefault();
+
+  if (!weaponPicked) {
+    // This will remove the reveal the crime message if any
+    cleanRevealMysteryMessage();
+
+    toggleAnimationDeckCard("weaponCard");
+
+    // This will randomly select a weapon from the weapons. And add that to the mystery object.
+    setTimeout(() => {
+      mystery.weapon = randomSelector(weapons);
+
+      // This will change the card to the one connected to the chosen weapon and show the name and weight of the weapon.
+      document.getElementById(
+        "weaponName"
+      ).innerHTML = `${mystery.weapon.name}`;
+      document.getElementById(
+        "weaponWeight"
+      ).innerHTML = `Weight: ${mystery.weapon.weight} lbs`;
+    }, 2000);
+    weaponPicked = true;
+  }
+};
+
+// This function will be invoked when you click on the room card.
+const pickRoom = (event) => {
+  event.preventDefault();
+
+  if (!roomPicked) {
+    // This will remove the reveal the crime message if any
+    cleanRevealMysteryMessage();
+
+    toggleAnimationDeckCard("roomCard");
+
+    // This will randomly select a room from the rooms. And add that to the mystery object.
+    setTimeout(() => {
+      mystery.room = randomSelector(rooms);
+
+      // This will change the card to the one connected to the chosen room and show which room it is.
+      document.getElementById("roomName").innerHTML = `${mystery.room}`;
+    }, 2000);
+    roomPicked = true;
+  }
+};
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+const revealMystery = (event) => {
+  event.preventDefault();
+  if (
+    mystery.killer === null ||
+    mystery.weapon === null ||
+    mystery.room === null
+  ) {
+    document.getElementById("mystery").innerHTML =
+      "Click all cards above to reveal the crime!";
+  } else {
+    document.getElementById(
+      "mystery"
+    ).innerHTML = `The murder was commited by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} room with a ${mystery.weapon.name}!`;
+  }
+};
+
+document.getElementById("killerCard").addEventListener("click", pickKiller);
+document.getElementById("weaponCard").addEventListener("click", pickWeapon);
+document.getElementById("roomCard").addEventListener("click", pickRoom);
+document.getElementById("revealCard").addEventListener("click", revealMystery);
```

Comment:
This looks super clean! I struggled to understand the syntax of `.addEventListener`